lib/big_query.ex

```diff
@@ -1,22 +1,33 @@
 defmodule BigQuery do
 
-  defdelegate podcasts(), to: BigQuery.Podcasts, as: :list
-  defdelegate podcast(id), to: BigQuery.Podcasts, as: :show
+  alias BigQuery.Podcasts, as: Podcasts
+  alias BigQuery.Episodes, as: Episodes
+  alias BigQuery.Downloads, as: Downloads
+  alias BigQuery.Impressions, as: Impressions
 
-  defdelegate episodes(), to: BigQuery.Episodes, as: :list
-  defdelegate episode(guid), to: BigQuery.Episodes, as: :show
+  defmodule Interval do
+    @enforce_keys [:from, :to, :seconds]
+    defstruct [:from, :to, :seconds]
+  end
 
-  defdelegate podcast_downloads(id, from, to, interval),
-    to: BigQuery.Downloads,
-    as: :for_podcast
-  defdelegate episode_downloads(guid, from, to, interval),
-    to: BigQuery.Downloads,
-    as: :for_episode
+  defmodule Grouping do
+    @enforce_keys [:name, :table, :key, :display, :fkey, :limit]
+    defstruct [:name, :table, :key, :display, :fkey, limit: 10]
+  end
 
-  defdelegate podcast_impressions(id, from, to, interval),
-    to: BigQuery.Impressions,
-    as: :for_podcast
-  defdelegate episode_impressions(guid, from, to, interval),
-    to: BigQuery.Impressions,
-    as: :for_episode
+  defdelegate podcasts(), to: Podcasts, as: :list
+  defdelegate podcast(id), to: Podcasts, as: :show
+
+  defdelegate episodes(), to: Episodes, as: :list
+  defdelegate episode(guid), to: Episodes, as: :show
+
+  defdelegate podcast_downloads(id, interval), to: Downloads, as: :for_podcast
+  defdelegate podcast_downloads(id, interval, group), to: Downloads, as: :for_podcast
```

Comment:
See, now `interval` and `group` are structs!  Much shorter than `podcast_downloads(id, from, to, interval_s, group_name, group_limit)`.